modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/AbstractAggregatePlannerTest.java

```diff
@@ -847,6 +850,20 @@ enum TestCase {
          * <p>Distribution identity(1)
          */
         CASE_23C("SELECT val0, AVG(val1) FROM test GROUP BY val0", schema(identity(1))),
+
+        /**
+         * Query: SELECT COUNT(val0), COUNT(DISTINCT(val1) from test.
+         *
+         * <p>Distribution single()
+         */
+        CASE_24_1("SELECT COUNT(val0), COUNT(DISTINCT(val1)) from test", schema(single())),
+
+        /**
+         * Query: SELECT COUNT(val0), COUNT(DISTINCT(val1) from test.
+         *
+         * <p>Distribution hash(0)
+         */
+        CASE_24_1A("SELECT COUNT(val0), COUNT(DISTINCT(val1)) from test", schema(hash()))
```

Comment:
Let's also check plans for hash distribution: hash(val1)

Reply:
Added test cases for both  hash(val0) and hash(val1).